R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,447 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
 
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L2322.SubsectorInterpTo_Fert %>%
+      add_title("Subsector shareweight interpolation of fertilizer sector") %>%
+      add_units("NA") %>%
+      add_comments("Created as empty table") %>%
       add_legacy_name("L2322.SubsectorInterpTo_Fert") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("energy/A322.subsector_interp", "common/GCAM_region_names") ->
       L2322.SubsectorInterpTo_Fert
 
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L2322.StubTech_Fert %>%
+      add_title("Identification of stub technologies of fertilizer sector") %>%
+      add_units("NA") %>%
+      add_comments("Expanded identification of stub technologies of fertilizer sector to all GCAM regions") %>%
       add_legacy_name("L2322.StubTech_Fert") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("energy/A322.globaltech_shrwt", "common/GCAM_region_names") ->
       L2322.StubTech_Fert
 
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L2322.GlobalTechShrwt_Fert %>%
+      add_title("Shareweights of global fertilizer sector technologies") %>%
+      add_units("Unitless") %>%
+      add_comments("Interpolated orginal data into all model years") %>%
       add_legacy_name("L2322.GlobalTechShrwt_Fert") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("energy/A322.globaltech_shrwt") ->
       L2322.GlobalTechShrwt_Fert
 
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
-      add_legacy_name("L2322.GlobalTechCoef_Fert") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+    L2322.GlobalTechCoef_Fert %>%
+      add_title("Energy inputs and coefficients of global fertilizer energy use and feedstocks technologies") %>%
+      add_units("Unitless") %>%
```

Comment:
I doubt these are unitless

Reply:
Should be "GJ per kg N"

Reply:
So is it in general that the `coefficient` should be having a unit?  

The `coefficient` in `L2322.GlobalTechCoef_Fert` came from `A322.globaltech_coef` which doesn't have much information before I added it into data folder. I also randomly checked a few other `A*_coef.csv` files in the new data system and they all have 'Unitless' in comments. 

Apologies in advance if I totally get it wrong.

Reply:
[Learning opportunity!] With the exception of the infamous "tranTechnology", in GCAM the "coefficient" of any technology and input simply represents the ratio between units of input and units of output. The units of the "output" of any technology are indicated in the tag "output-unit" specified in the corresponding "A\*.sector.csv" file. For the most part, this is also a reliable source of the "input-unit" though in some cases, different inputs may have different units (e.g., cement takes inputs of limestone (Mt) and heat (EJ)). The fail-safe way of determining the model's units of any input is to find the supplysector or resouce of the same name and check its output unit. In this case, fertilizer's output unit is Mt and inputs are EJ, so the (input-output) coefficient is EJ/Mt, which we simplify by power of 9 to GJ/kg. GCAM's standard of simplifying indicators by powers of 9 comes from the order of magnitude of the global population.

Reply:
I thought this was useful enough that I put it [into the wiki](https://github.com/JGCRI/gcamdata/wiki/Units).

Reply:
Thanks! @pkyle 

@bpbond Turns out in the `L2322.GlobalTechCoef_Fert` there would be two kinds of unit existing: GJ/kg N for supplysector N fertilizer and Unitless for supplysector Exports_fertilizer. 

Does the test accept something like?

```
add_units("GJ/kg N for supplysector N fertilizer; Unitless for supplysector Exports_fertilizer")
```

Reply:
Sure. The test just verifies that you've entered _something_ for units.